client-app/shared/static-content/components/related-products.vue

```diff
@@ -1,59 +1,85 @@
 <template>
   <!-- Related products section -->
-  <div v-show="relatedProducts.length" class="mt-5 flex flex-col print:hidden lg:flex-row">
-    <div class="-mx-4.5 print:mx-0 print:grow lg:mx-0 lg:w-full xl:w-9/12 xl:pe-8">
-      <VcWidget :title="$t('pages.product.related_product_section_title')" prepend-icon="cube" size="lg">
-        <div v-if="isMobile" class="-mb-2 flex flex-wrap items-stretch gap-x-7 gap-y-4 px-2">
-          <ProductCardRelated
-            v-for="(item, index) in relatedProducts.slice(0, 4)"
-            :key="index"
-            :product="item"
-            class="w-[calc((100%-1.75rem)/2)] sm:w-[calc((100%-2*1.75rem)/3)] md:w-[calc((100%-3*1.75rem)/4)]"
-            @link-click="ga.selectItem(item)"
-          />
-        </div>
-
-        <VcCarousel v-else :slides="relatedProducts" :options="relatedProductsCarouselOptions" navigation>
-          <template #slide="{ slide: item }">
-            <div class="h-full px-2 py-3">
-              <ProductCardRelated class="h-full" :product="item" @link-click="ga.selectItem(item)" />
-            </div>
-          </template>
-        </VcCarousel>
-      </VcWidget>
+  <VcWidget
+    v-if="relatedProducts.length"
+    :title="$t('pages.product.related_product_section_title')"
+    prepend-icon="cube"
+    size="lg"
+    class="order-last print:hidden max-md:-mx-4.5"
+  >
+    <div v-if="lg" class="-mb-2 flex flex-wrap items-stretch gap-x-7 gap-y-4">
+      <ProductCardRelated
+        v-for="(item, index) in mobileProducts"
+        :key="index"
+        :product="item"
+        class="w-[calc((100%-1.75rem)/2)] sm:w-[calc((100%-2*1.75rem)/3)] md:w-[calc((100%-1.75rem)/2)]"
+        @link-click="ga.selectItem(item)"
```

Comment:
Let's move it to separate handler and call GA inside, then reuse it here and below

Reply:
https://virtocommerce.atlassian.net/browse/ST-5654